### PR TITLE
SEAR-1239 Add optional CORS middleware for HTTP servers

### PR DESCRIPTION
**Pull request by theisenmark**
**Issue Link**
https://spothero.atlassian.net/browse/SEAR-1239

**Description**
Adds optional CORS middleware for HTTP servers.

### CODE REVIEW

File: cors/cli.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+package cors
+
+import "github.com/spf13/pflag"
+
+// RegisterFlags registers JOSE flags with pflags
+func (c *Config) RegisterFlags(flags *pflag.FlagSet) {
+	flags.BoolVar(&c.EnableMiddleware, "cors-enable-middleware", c.EnableMiddleware, "Specify whether or not CORS middleware is enabled to  enforce policies on cross origin requests")
+	flags.StringVar(&c.AllowedOrigins, "cors-allowed-origins", c.AllowedOrigins, "Specify which origin(s) allow responses to be populated with headers to allow cross origin requests (e.g. \"*\") or \"https://example.com\")")
+	flags.StringVar(&c.AllowedMethods, "cors-allowed-methods", c.AllowedMethods, "Specify which method(s) allow responses to be populated with headers to allow cross origin requests (e.g. \"POST, GET, OPTIONS, PUT, DELETE\")")
+	flags.StringVar(&c.AllowedMethods, "cors-allowed-headers", c.AllowedHeaders, "Specify which header(s) which headers are allowed for cross origin requests (e.g. \"*\" or \"Accept, Content-Type, Content-Length\")")
```
Comment (greg-mclain) [inline cors/cli.go:24]:
```suggestion
	flags.StringVar(&c.AllowedHeaders, "cors-allowed-headers", c.AllowedHeaders, "Specify which header(s) are allowed for cross origin requests (e.g. \"*\" or \"Accept, Content-Type, Content-Length\")")
```

**Review by greg-mclain — APPROVED** [review]
One minor change needed, otherwise looks good to me! Nice work!

File: cors/cli.go (hunk 1 above)
Comment (theisenmark) [reply cors/cli.go:24]: Good catch, thanks @greg-mclain

**Pull request merged**